pkg/geo/geomfn/subdivide_test.go

```diff
@@ -0,0 +1,216 @@
+				geo.MustParseGeometry("LINESTRING(7.5 5,15 10)"),
+				geo.MustParseGeometry("LINESTRING(15 10,15 0)"),
+			},
+		},
+		{
+			"LineString with specified SRID",
+			args{geo.MustParseGeometry("SRID=4269;LINESTRING(0 0, 10 15, 0 0, 10 15, 10 0, 10 15)"), 5},
+			[]geo.Geometry{
+				geo.MustParseGeometry("SRID=4269;LINESTRING(0 0,5 7.5)"),
+				geo.MustParseGeometry("SRID=4269;LINESTRING(10 7.5,10 0)"),
+				geo.MustParseGeometry("SRID=4269;LINESTRING(5 7.5,10 15)"),
+				geo.MustParseGeometry("SRID=4269;LINESTRING(10 15,10 7.5)"),
+			},
+		},
+		{
+			"MultiLineString - horizontal and vertical lines",
+			args{geo.MustParseGeometry("MULTILINESTRING((5 0, 5 1, 5 3, 5 4, 5 5, 5 6),(0 5, 1 5, 2 5, 3 5, 4 5, 6 5))"), 5},
+			[]geo.Geometry{
+				geo.MustParseGeometry("LINESTRING(5 0,5 3)"),
+				geo.MustParseGeometry("LINESTRING(5 3,5 6)"),
+				geo.MustParseGeometry("LINESTRING(0 5,3 5)"),
+				geo.MustParseGeometry("LINESTRING(3 5,6 5)"),
+			},
+		},
+		{
+			"MultiPoint, max vertices 6",
+			args{geo.MustParseGeometry("MULTIPOINT((0 1),(1 2),(2 3),(3 4),(4 3),(6 2),(7 1),(8 0),(9 -1),(10 -2),(11 -3))"), 6},
+			[]geo.Geometry{
+				geo.MustParseGeometry("MULTIPOINT(0 1,1 2,2 3,3 4,4 3)"),
+				geo.MustParseGeometry("MULTIPOINT(6 2,7 1,8 0,9 -1,10 -2,11 -3)"),
+			},
+		},
+		{
+			"GeometryCollection, types with different dimensions",
+			args{geo.MustParseGeometry("GEOMETRYCOLLECTION(LINESTRING(0 0, 10 10, 0 10, 10 0), POLYGON((0 0, -2 -1, -2 1, 0 0, 2 1, 2 -1, 0 0)))"), 5},
+			[]geo.Geometry{
+				geo.MustParseGeometry("POLYGON((0 0,-2 -1,-2 1,0 0))"),
+				geo.MustParseGeometry("POLYGON((0 0,2 1,2 -1,0 0))"),
+			},
+		},
+		{
+			"GeometryCollection, types with different dimensions, point included",
+			args{geo.MustParseGeometry("GEOMETRYCOLLECTION(LINESTRING(0 0, 10 10, 0 10, 10 0), POINT(1 10))"), 5},
+			[]geo.Geometry{
+				geo.MustParseGeometry("LINESTRING(0 0,10 10,0 10,10 0)"),
+			},
+		},
+		{
+			"GeometryCollection, same types, one invalid",
+			args{geo.MustParseGeometry("GEOMETRYCOLLECTION(POLYGON((0 0, -2 -1, -2 1, 0 0, 2 1, 2 -1, 0 0)),POLYGON((-1 -1,-1 -0.5, -1 0, 1 0.5, 1 -1,-1 -1)))"), 5},
+			[]geo.Geometry{
+				geo.MustParseGeometry("POLYGON((0 0,-2 -1,-2 1,0 0))"),
+				geo.MustParseGeometry("POLYGON((0 0,2 1,2 -1,0 0))"),
+				geo.MustParseGeometry("POLYGON((-1 -1,-1 0,0 0.25,0 -1,-1 -1))"),
+				geo.MustParseGeometry("POLYGON((0 0.25, 1 0.5, 1 -1, 0 -1, 0 0.25))"),
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := Subdivide(tt.args.g, tt.args.maxVertices)
+			require.NoError(t, err)
+			require.Equal(t, tt.want, got)
+		})
+	}
+
+	t.Run("less max vertices than minimum acceptable", func(t *testing.T) {
+		_, err := Subdivide(geo.MustParseGeometry("LINESTRING(0 0, -10 10, 0 10)"), 4)
+		require.Error(t, err)
+		require.Equal(t, "max_vertices number cannot be less than 5", err.Error())
+	})
+
+	t.Run("max number of division exceeded", func(t *testing.T) {
```

Comment:
nit: maybe `subdivide would need to divide more than the maximum depth`

Reply:
Fixed